src/pyodide/python-entrypoint-helper.js

```diff
@@ -210,16 +210,28 @@ async function setupPackages(pyodide) {
   return pyodide.pyimport(mainModule);
 }
 
+let mainModulePromise;
+function getMainModule() {
+  if (mainModulePromise !== undefined) {
+    return mainModulePromise;
+  }
+  mainModulePromise = (async function() {
+    // TODO: investigate whether it is possible to run part of loadPyodide in top level scope
+    // When we do it in top level scope we seem to get a broken file system.
+    const pyodide = await loadPyodide();
+    return await setupPackages(pyodide);
+  })();
+  return mainModulePromise;
```

Comment:
Bit easier with at least one fewer Promise allocated...

```suggestion
  mainModulePromise = setupPackages(await loadPyodide());
  return mainModulePromise;
```

Reply:
Is this still guaranteed to only execute once? As I understand it, 
```js
mainModulePromise = setupPackages(await loadPyodide());
```
means the same thing as:
```js
const tmp = await loadPyodide();
mainModulePromise = setupPackages(tmp);
```
and so there is an await before assigning to `mainModulePromise` meaning that a second request could come in and observe `mainModulePromise` as not null. The way I wrote it with the outside function being synchronous and assigning to the promise guarantees that we don't yield to the event loop when `mainModulePromise` is still `undefined.

Reply:
hah... doh you're right lol. that's what I get for doing code review at dinner time.